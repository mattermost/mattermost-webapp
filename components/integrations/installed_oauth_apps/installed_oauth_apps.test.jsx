// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import InstalledOAuthApps from 'components/integrations/installed_oauth_apps/installed_oauth_apps.jsx';
import BackstageList from 'components/backstage/components/backstage_list.jsx';

describe('components/integrations/InstalledOAuthApps', () => {
    const oauthApps = {
        facxd9wpzpbpfp8pad78xj75pr: {
            id: 'facxd9wpzpbpfp8pad78xj75pr',
            name: 'firstApp',
            client_secret: '88cxd9wpzpbpfp8pad78xj75pr',
            create_at: 1501365458934,
            creator_id: '88oybd1dwfdoxpkpw1h5kpbyco',
            description: 'testing',
            homepage: 'https://test.com',
            icon_url: 'https://test.com/icon',
            is_trusted: false,
            update_at: 1501365458934,
            callback_urls: ['https://test.com/callback'],
        },
        fzcxd9wpzpbpfp8pad78xj75pr: {
            id: 'fzcxd9wpzpbpfp8pad78xj75pr',
            name: 'secondApp',
            client_secret: 'decxd9wpzpbpfp8pad78xj75pr',
            create_at: 1501365459984,
            creator_id: '88oybd2dwfdoxpkpw1h5kpbyco',
            description: 'testing2',
            homepage: 'https://test2.com',
            icon_url: 'https://test2.com/icon',
            is_trusted: true,
            update_at: 1501365479988,
            callback_urls: ['https://test2.com/callback', 'https://test2.com/callback2'],
        },
    };

    const baseProps = {
        team: {name: 'test'},
        oauthApps,
        canManageOauth: true,
        regenOAuthAppSecretRequest: {
            status: 'not_started',
            error: null,
        },
        actions: {
            getOAuthApps: jest.fn(),
            regenOAuthAppSecret: jest.fn(),
            deleteOAuthApp: jest.fn(),
        },
        enableOAuthServiceProvider: true,
    };

    test('should match snapshot', () => {
        const newGetOAuthApps = jest.fn().mockImplementation(
            () => {
                return new Promise((resolve) => {
                    process.nextTick(() => resolve());
                });
            }
        );

        const props = {...baseProps};
        props.actions.getOAuthApps = newGetOAuthApps;
        const wrapper = shallow(
            <InstalledOAuthApps {...props}/>
        );
        expect(wrapper.find('InstalledOAuthApp').length).toBe(2);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(BackstageList).props().addLink).toEqual('/test/integrations/oauth2-apps/add');
        expect(wrapper.find(BackstageList).props().addText).toEqual('Add OAuth 2.0 Application');

        wrapper.setProps({canManageOauth: false});
        expect(wrapper.find(BackstageList).props().addLink).toBeFalsy();
        expect(wrapper.find(BackstageList).props().addText).toBeFalsy();
    });

    test('should props.deleteOAuthApp on deleteOAuthApp', () => {
        const newDeleteOAuthApp = jest.fn();
        const props = {...baseProps};
        props.actions.deleteOAuthApp = newDeleteOAuthApp;
        const wrapper = shallow(
            <InstalledOAuthApps {...props}/>
        );

        wrapper.instance().deleteOAuthApp(oauthApps.facxd9wpzpbpfp8pad78xj75pr);
        expect(newDeleteOAuthApp).toHaveBeenCalled();
        expect(newDeleteOAuthApp).toHaveBeenCalledWith(oauthApps.facxd9wpzpbpfp8pad78xj75pr.id);
    });
});
