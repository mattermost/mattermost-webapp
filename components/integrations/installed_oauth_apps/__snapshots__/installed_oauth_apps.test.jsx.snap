// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/integrations/InstalledOAuthApps should match snapshot 1`] = `
<BackstageList
  addButtonId="addOauthApp"
  addLink="/test/integrations/oauth2-apps/add"
  addText="Add OAuth 2.0 Application"
  emptyText={
    <FormattedMessage
      defaultMessage="No OAuth 2.0 Applications found"
      id="installed_oauth_apps.empty"
      values={Object {}}
    />
  }
  emptyTextSearch={
    <injectIntl(FormattedMarkdownMessage)
      defaultMessage="No OAuth 2.0 Applications match {searchTerm}"
      id="installed_oauth_apps.emptySearch"
    />
  }
  header={
    <FormattedMessage
      defaultMessage="OAuth 2.0 Applications"
      id="installed_oauth_apps.header"
      values={Object {}}
    />
  }
  helpText={
    <FormattedMessage
      defaultMessage="Create {oauthApplications} to securely integrate bots and third-party apps with Mattermost. Visit the {appDirectory} to find available self-hosted apps."
      id="installed_oauth_apps.help"
      values={
        Object {
          "appDirectory": <a
            href="https://about.mattermost.com/default-app-directory/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <FormattedMessage
              defaultMessage="App Directory"
              id="installed_oauth_apps.help.appDirectory"
              values={Object {}}
            />
          </a>,
          "oauthApplications": <a
            href="https://docs.mattermost.com/developer/oauth-2-0-applications.html"
            rel="noopener noreferrer"
            target="_blank"
          >
            <FormattedMessage
              defaultMessage="OAuth 2.0 applications"
              id="installed_oauth_apps.help.oauthApplications"
              values={Object {}}
            />
          </a>,
        }
      }
    />
  }
  loading={true}
  searchPlaceholder="Search OAuth 2.0 Applications"
>
  <Component />
</BackstageList>
`;

exports[`components/integrations/InstalledOAuthApps should match snapshot 2`] = `
<div>
  <Connect(InstalledOAuthApp)
    key="facxd9wpzpbpfp8pad78xj75pr"
    oauthApp={
      Object {
        "callback_urls": Array [
          "https://test.com/callback",
        ],
        "client_secret": "88cxd9wpzpbpfp8pad78xj75pr",
        "create_at": 1501365458934,
        "creator_id": "88oybd1dwfdoxpkpw1h5kpbyco",
        "description": "testing",
        "homepage": "https://test.com",
        "icon_url": "https://test.com/icon",
        "id": "facxd9wpzpbpfp8pad78xj75pr",
        "is_trusted": false,
        "name": "firstApp",
        "update_at": 1501365458934,
      }
    }
    onDelete={[Function]}
    onRegenerateSecret={[MockFunction]}
    regenOAuthAppSecretRequest={
      Object {
        "error": null,
        "status": "not_started",
      }
    }
    team={
      Object {
        "name": "test",
      }
    }
  />
</div>
`;

exports[`components/integrations/InstalledOAuthApps should match snapshot 3`] = `<div />`;
